pulsar-client/src/test/java/org/apache/pulsar/client/impl/ConsumerImplTest.java

```diff
@@ -200,4 +201,21 @@ public void testClose() {
         }
         Assert.assertNull(checkException);
     }
+
+    @Test
+    public void testConsumerCreatedWhilePaused() throws InterruptedException {
```

Comment:
Do you think you could write a test that relies on the `ConsumerBuilder` implementation? That way we can exercise all of your changes.

Reply:
I added a test in the `ConsumerBuilderImplTest` class 👍🏻 

If you meant I should update _this_ test to use the builder - I can do that too! (though it's a little less natural with the way this test class is currently set up)

Reply:
I think it makes sense to keep both tests.